File: experiments/2026-09-19-1347/react/src/domain.ts
/**
 * The sales knowledge, embedded at build time.
 *
 * The blueprint allows embedding or fetching over HTTP. Embedding is what Vite
 * makes cheap here, and it keeps the app a static bundle with no backend.
 */

import knowledge from '../../../../domain/knowledge.md?raw';
import classic from '../../../../domain/landing_pages/classic.md?raw';
import eco from '../../../../domain/landing_pages/eco.md?raw';
import family from '../../../../domain/landing_pages/family.md?raw';
import mini from '../../../../domain/landing_pages/mini.md?raw';
import silent from '../../../../domain/landing_pages/silent.md?raw';
import slim from '../../../../domain/landing_pages/slim.md?raw';

export const KNOWLEDGE = knowledge;

export const LANDING_PAGES: Record<string, string> = {
  classic,
  eco,
  family,
  mini,
  silent,
  slim,
};

/** The knowledge base plus every landing page, as one block for the prompt. */
export function domainCorpus(): string {
  const pages = Object.entries(LANDING_PAGES)
    .map(([name, body]) => `--- landing page: ${name} ---\n${body}`)
    .join('\n\n');
  return `${KNOWLEDGE}\n\n${pages}`;
}

File: ci/experiments/2026-09-19-1347/jaspr/lib/domain.dart
// The sales knowledge, fetched over HTTP.
//
// The blueprint allows embedding or fetching. This arm fetches: `web/domain` is
// a symlink to the repo's domain folder, so the files are served next to the
// app and read at startup. It keeps the knowledge out of the compiled bundle,
// which means it can be changed without rebuilding.

import 'package:http/http.dart' as http;

const List<String> _pages = [
  'classic',
  'eco',
  'family',
  'mini',
  'silent',
  'slim',
];

Future<String> loadDomainCorpus() async {
  Future<String> fetch(String path) async {
    final res = await http.get(Uri.parse(path));
    if (res.statusCode != 200) {
      throw Exception('could not read $path (HTTP ${res.statusCode})');
    }
    return res.body;
  }

  final knowledge = await fetch('domain/knowledge.md');
  final pages = <String>[];
  for (final name in _pages) {
    pages.add(
      '--- landing page: $name ---\n'
      '${await fetch('domain/landing_pages/$name.md')}',
    );
  }
  return '$knowledge\n\n${pages.join('\n\n')}';
}
